scripts/ag

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+
+import click
+import os
+import json
+from gzip import open as gzopen
+
+from biom import load_table
+
+from americangut.taxtree import (build_tree_from_taxontable,
+                                 sample_rare_unique,
+                                 build_persample_tree_from_taxontable)
+
+
+@click.group()
+def ag():
+    pass
+
+
+@ag.group()
+def taxtree():
+    pass
+
+
+@taxtree.command()
+@click.option('--table', required=True, type=click.Path(exists=True),
+              help='The input BIOM taxon table')
+@click.option('--rare-threshold', required=False,
+              default=0.01, help='The threshold to consider an OTU rare')
+def rareunique(table, rare_threshold):
+    table = load_table(table)
+    tree, all_taxa = build_tree_from_taxontable(table)
+    click.echo("#SampleID\ttype\ttaxon")
+    it = sample_rare_unique(tree, None, all_taxa, rare_threshold)
+    for samp, ft, rare, uniq in it:
+        for r in rare:
+            click.echo("%s\t%s\t%s" % (samp, 'rare', r))
+        for u in uniq:
+            click.echo("%s\t%s\t%s" % (samp, 'unique', u))
+
+
+@taxtree.command()
+@click.option('--table', required=True, type=click.Path(exists=True),
+              help='The input BIOM taxon table')
+@click.option('--output', required=True, type=click.Path(exists=False))
+def persample_json(table, output):
+    os.makedirs(output)
```

Comment:
that is surprising, good catch

On Wed, Mar 25, 2015 at 10:42 AM, adamrp <notifications@github.com> wrote:

> In scripts/ag
> <https://github.com/biocore/American-Gut/pull/137#discussion_r27139627>:
>
> > +    tree, all_taxa = build_tree_from_taxontable(table)
> > +    click.echo("#SampleID\ttype\ttaxon")
> > +    it = sample_rare_unique(tree, None, all_taxa, rare_threshold)
> > +    for samp, ft, rare, uniq in it:
> > +        for r in rare:
> > +            click.echo("%s\t%s\t%s" % (samp, 'rare', r))
> > +        for u in uniq:
> > +            click.echo("%s\t%s\t%s" % (samp, 'unique', u))
> > +
> > +
> > +@taxtree.command()
> > +@click.option('--table', required=True, type=click.Path(exists=True),
> > +              help='The input BIOM taxon table')
> > +@click.option('--output', required=True, type=click.Path(exists=False))
> > +def persample_json(table, output):
> > +    os.makedirs(output)
>
> I didn't know this, but apparently makedirs
> <https://docs.python.org/2/library/os.html#os.makedirs> gets confused if
> the argument contains os.pardir
> <https://docs.python.org/2/library/os.html#os.pardir> (so, .. probably).
> This seems like it could happen fairly regularly, so maybe call
> os.path.abspath on output before os.makedirs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/American-Gut/pull/137/files#r27139627>.
>